packages/api-gateway/src/services/validators.ts

```diff
@@ -71,17 +83,16 @@ export function assertAvailableDescriptorExists(
   }
 }
 
-export function assertNonDraftDescriptor(
+export function assertNotValidDescriptor(
```

Comment:
This assertions asserts that the descriptor **is valid** as also the return type signature states - it should be renamed to `assertValidDescriptor` or `assertIsValidDescriptor`

```suggestion
export function assertIsValidDescriptor(
```